src/main/User/UserController.java

```diff
@@ -45,6 +54,49 @@ public UserController(MongoDatabase db) {
     logger = l.createLogger("UserController");
   }
 
+  public InputStream getUserPfp(String username) {
+    Bson filter = Filters.eq("metadata.owner", username);
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSFile grid_out = gridBucket.find(filter).first();
+    if (grid_out == null || grid_out.getMetadata() == null) {
+      return null;
+    }
+    InputStream pfp = gridBucket.openDownloadStream(grid_out.getObjectId());
+    return pfp;
+  }
+
+  public void uploadPfp(UploadedFile file, String username) {
+    String fileName = file.getFilename();
+    InputStream content = file.getContent();
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSUploadOptions options =
+        new GridFSUploadOptions()
+            .chunkSizeBytes(100000)
+            .metadata(
+                new Document("type", "pfp")
+                    .append("upload_date", String.valueOf(LocalDate.now()))
+                    .append("owner", username));
+    gridBucket.uploadFromStream(fileName, content, options);
+  }
+
+  public Handler uploadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        logger.info(username + " is Attempting to upload a profile picture");
+        UploadedFile file = ctx.uploadedFile("file");
+        uploadPfp(file, username);
```

Comment:
Does this function ever throw an error? Is it possible to unsuccessfully upload a profile pic? Same comment below for loading one.